src/System.Private.CoreLib/src/System/Exception.CoreCLR.cs

```diff
@@ -126,9 +127,10 @@ internal bool TryGetRestrictedLanguageErrorObject(out object restrictedErrorObje
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         static private extern IRuntimeMethodInfo GetMethodFromStackTrace(object stackTrace);
 
-        private MethodBase GetExceptionMethodFromStackTrace()
+        private MethodBase? GetExceptionMethodFromStackTrace()
         {
-            IRuntimeMethodInfo method = GetMethodFromStackTrace(_stackTrace);
+            // _stackTrace shouldn't be null because we only call this method when it is not null.
```

Comment:
Can you put this into a Debug.Assert instead of a comment, e.g.
```C#
Debug.Assert(_stackTrace != null);
```
?  You could add a message to the assert if desired.

Reply:
Sure, makes sense 😄